pkg/kn/commands/types.go

```diff
@@ -69,6 +72,10 @@ func (params *KnParams) Initialize() {
 		params.NewSourcesClient = params.newSourcesClient
 	}
 
+	if params.NewEventingClient == nil {
+		params.NewEventingClient = params.newEventingClient
+	}
```

Comment:
This reads odd... would make sense as a lazy init but that would mean that the sentence inside the `if` would make a function call to create not just an assignment which now means that the right hand side has to be created before...

Reply:
This technique allows test code to hook in by setting the `NewEventingClient` in advance before reaching this method with a fake implementation.

I'm also not happy with this construct (for other reasons, mostly because of coupling layers unnecessarily), but I'm happy to take this over as is (as its follows the same scheme like for the other clients).

In a subsequent PR we should refactor the client handling to be more delocalized maybe.

Reply:
Agree. Let's refactor them in a separate PR.